### PR TITLE
Improve user experience for VCS metadata generator menu button to mitigate accidentally overriding

### PR DESCRIPTION
**Pull request by Cass-dev-web**
Simply added a check to see if Git is already initialized, and if so, it would change the user interface to display `Override Version Control Metadata...` instead of `Create Version Control Metadata...` to avoid confusion (just a polish, low priority).

Please review the code and provide feedback as I am new to push requests.

### CODE REVIEW

**Comment by Cass-dev-web** [comment]
Commit [6416791](https://github.com/godotengine/godot/pull/88609/commits/6416791f826cea99dbcc6ba154cdb1513a96406c): Added a wacky hack which checks that the class name does not conflict with `Editor` as it would cause #88601 to happen. 

**Needs work & needs to be tested for non-node extending scripts.**

File: modules/gdscript/gdscript_analyzer.cpp [hunk 1]
```diff
@@ -5602,7 +5602,7 @@ void GDScriptAnalyzer::resolve_pending_lambda_bodies() {
 }
 
 bool GDScriptAnalyzer::class_exists(const StringName &p_class) const {
-	return ClassDB::class_exists(p_class) && ClassDB::is_class_exposed(p_class);
+	return (ClassDB::class_exists(p_class) && ClassDB::is_class_exposed(p_class)) || p_class=="Editor";
```
Comment (dalexeev) [inline modules/gdscript/gdscript_analyzer.cpp:5605]: This is incorrect. The issue is about displaying **custom** classes in the editor dialog, and you are trying to fix it in the analyzer method that checks **native** classes. Also, this is unrelated to the PR title.

File: editor/editor_node.cpp [hunk 2]
```diff
@@ -7361,7 +7361,9 @@ EditorNode::EditorNode() {
 	vcs_actions_menu = VersionControlEditorPlugin::get_singleton()->get_version_control_actions_panel();
 	vcs_actions_menu->set_name("Version Control");
 	vcs_actions_menu->connect("index_pressed", callable_mp(this, &EditorNode::_version_control_menu_option));
-	vcs_actions_menu->add_item(TTR("Create Version Control Metadata..."), RUN_VCS_METADATA);
+	Ref<FileAccess> fileAccess = FileAccess::create(FileAccess::ACCESS_RESOURCES);
+	String isGitInitialisedSymbol = (fileAccess->file_exists(current_path.path_join(".gitignore")) ? "Override" : "Create");
+	vcs_actions_menu->add_item(TTR(isGitInitialisedSymbol + " Version Control Metadata..."), RUN_VCS_METADATA);
```
Comment (dalexeev) [inline editor/editor_node.cpp:7366]: `TTR()` must contain a string literal, not an expression. Otherwise, our script will not be able to extract the string for translation.

**Comment by Cass-dev-web** [comment]
Updated

File: editor/editor_node.cpp [hunk 3]
```diff
@@ -7361,7 +7361,11 @@ EditorNode::EditorNode() {
 	vcs_actions_menu = VersionControlEditorPlugin::get_singleton()->get_version_control_actions_panel();
 	vcs_actions_menu->set_name("Version Control");
 	vcs_actions_menu->connect("index_pressed", callable_mp(this, &EditorNode::_version_control_menu_option));
-	vcs_actions_menu->add_item(TTR("Create Version Control Metadata..."), RUN_VCS_METADATA);
+	Ref<FileAccess> fileAccess = FileAccess::create(FileAccess::ACCESS_RESOURCES);
```
Comment (DarkMessiah) [inline editor/editor_node.cpp:7364]:
```suggestion
	Ref<FileAccess> file_access = FileAccess::create(FileAccess::ACCESS_RESOURCES);
```
You should use `snake_case` for the code.

File: editor/editor_node.cpp [hunk 4]
```diff
@@ -7361,7 +7361,11 @@ EditorNode::EditorNode() {
...
+	String isGitInitialisedSymbol = (fileAccess->file_exists(current_path.path_join(".gitignore"))
```
Comment (DarkMessiah) [inline editor/editor_node.cpp:7365]:
```suggestion
	String is_git_initialised_symbol = (fileAccess->file_exists(current_path.path_join(".gitignore"))
```
Same here :)

File: editor/editor_node.cpp [hunk 5]
```diff
@@ -7361,7 +7361,11 @@ EditorNode::EditorNode() {
 	vcs_actions_menu = VersionControlEditorPlugin::get_singleton()->get_version_control_actions_panel();
 	vcs_actions_menu->set_name("Version Control");
 	vcs_actions_menu->connect("index_pressed", callable_mp(this, &EditorNode::_version_control_menu_option));
-	vcs_actions_menu->add_item(TTR("Create Version Control Metadata..."), RUN_VCS_METADATA);
+	Ref<FileAccess> file_access = FileAccess::create(FileAccess::ACCESS_RESOURCES);
+	String is_git_initialised_symbol = (fileAccess->file_exists(current_path.path_join(".gitignore"))
+		? TTR("Override Version Control Metadata...")
+		: TTR("Create Version Control Metadata..."));
+	vcs_actions_menu->add_item(isGitInitialisedSymbol, RUN_VCS_METADATA);
```
Comment (dalexeev) [inline editor/editor_node.cpp:7368]:
```suggestion
	String vcs_metadata_item = FileAccess::exists("res://.gitignore") ? TTR("Override Version Control Metadata...") : TTR("Create Version Control Metadata...");
	vcs_actions_menu->add_item(vcs_metadata_item, RUN_VCS_METADATA);
```

Or maybe just:

```suggestion
	vcs_actions_menu->add_item(TTR("Setup Version Control Metadata..."), RUN_VCS_METADATA);
```

**Review by dalexeev — COMMENTED** [review]
This works, but the text is updated only after restarting the editor. This can be solved by checking the file on the `EditorFileSystem.filesystem_changed` signal or every time the popup is shown. In my opinion, this is an overkill, replacing "Create" with "Create/Override" or "Setup" will work too.

Also, please [squash commits into one](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase). To do this you need a local git client, GitHub web interface doesn't support it.

**Comment by Cass-dev-web** [comment]
> Also, please [squash commits into one](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase).
Squashed.

File: editor/editor_node.cpp [hunk 6]
```diff
@@ -7361,7 +7361,7 @@ EditorNode::EditorNode() {
 	vcs_actions_menu = VersionControlEditorPlugin::get_singleton()->get_version_control_actions_panel();
 	vcs_actions_menu->set_name("Version Control");
 	vcs_actions_menu->connect("index_pressed", callable_mp(this, &EditorNode::_version_control_menu_option));
-	vcs_actions_menu->add_item(TTR("Create Version Control Metadata..."), RUN_VCS_METADATA);
+	vcs_actions_menu->add_item(TTR("Create/Override Version Control Settings..."), RUN_VCS_METADATA);
```
Comment (dalexeev) [inline editor/editor_node.cpp:7364]: Note that the next item also contains "Settings":

Comment (Cass-dev-web) [reply editor/editor_node.cpp:7364]: Ah must've been a mistake, I didn't mean to write that. Will fix.

**Review by dalexeev — APPROVED** [review]
Looks good to me. Please squash commits one more time (note that you can do this locally before force-push to save CI resources).

**Comment by akien-mga** [comment]
You squashed it correctly, but adding "(squashed x2)" in the commit message was unnecessary. Once merged, it shouldn't matter to a reader whether that commit is the result of squashing multiple earlier commits or not, what matters is what the commit does :)

If you can amend the commit with `git commit --amend` to remove that part, it should be perfect. But if not it's not a big deal, we can also merge as is.

**Review by dalexeev — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!
